tests/core/vm/test_computation.py

```diff
@@ -0,0 +1,184 @@
+    # Get accounts for deletion can return multiple accounts
+    computation.msg.storage_address = CANONICAL_ADDRESS_2
+    computation.register_account_for_deletion(CANONICAL_ADDRESS_2)
+    accounts_for_deletion = sorted(computation.get_accounts_for_deletion(),
+                                   key=lambda item: item[0])
+    assert CANONICAL_ADDRESS_1 == accounts_for_deletion[0][0]
+    assert CANONICAL_ADDRESS_2 == accounts_for_deletion[1][0]
+
+
+def test_add_log_entry(computation):
+    # Log entries starts empty
+    assert computation.log_entries == []
+    with pytest.raises(ValidationError):
+        # Fails if address isn't canonical
+        computation.add_log_entry("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6", [1, 2, 3], b'1')
+        # Fails if topic elements aren't uint256
+        computation.add_log_entry(CANONICAL_ADDRESS_1, [-1, 2, 3], b'1')
+        computation.add_log_entry(CANONICAL_ADDRESS_1, ['1', 2, 3], b'1')
+        # Fails if data isn't bytes
+        computation.add_log_entry(CANONICAL_ADDRESS_1, [1, 2, 3], 1)
+    # Adds log entry to log entries
+    computation.add_log_entry(CANONICAL_ADDRESS_1, [1, 2, 3], b'1')
+    assert computation.log_entries == [(b'\x0fW.R\x95\xc5\x7f\x15\x88o\x9b&>/m-l{^\xc6',
+                                        [1, 2, 3],
+                                        b'1')]
+    # Can add multiple entries
+    computation.add_log_entry(CANONICAL_ADDRESS_1, [4, 5, 6], b'2')
+    computation.add_log_entry(CANONICAL_ADDRESS_1, [7, 8, 9], b'3')
+    assert len(computation.log_entries) == 3
+
+
+def test_get_log_entries(computation, create_error):
+    computation.add_log_entry(CANONICAL_ADDRESS_1, [1, 2, 3], b'1')
+    assert computation.log_entries == [(b'\x0fW.R\x95\xc5\x7f\x15\x88o\x9b&>/m-l{^\xc6',
+                                        [1, 2, 3],
+                                        b'1')]
+    # Trigger an out of gas error causing get log entries to be ()
+    computation = create_error(computation)
+    assert computation.get_log_entries() == ()
+
+
+def test_get_gas_refund(computation, create_error):
+    computation.gas_meter.gas_refunded = 100
+    assert computation.get_gas_refund() == 100
+    # Trigger an out of gas error causing get gas refund to be 0
+    computation = create_error(computation)
+    assert computation.get_gas_refund() == 0
+
+
+def test_output(computation, create_error):
+    computation.output = b'1'
+    assert computation.output == b'1'
+    # Trigger an out of gas error causing output to be b''
+    computation = create_error(computation)
+    assert computation.output == b''
+
+
+def test_get_gas_remaining(computation, create_error):
+    assert computation.get_gas_remaining() == 100
+    # Trigger an out of gas error causing get gas remaining to be 0
+    computation = create_error(computation)
+    assert computation.get_gas_remaining() == 0
+
+
+def test_get_gas_used(computation, create_error):
+    computation.msg.gas = 150
```

Comment:
This type of mutation is error prone.  It would be much better to have the computation be create with this value.  One way to do this would be to stop using a static `computation` fixture and instead use a *factory*.

```python
def _computation_factory(....<put-some-sane-defaults-here>):
    return Computation(<boiler-plate-goes-here>)
```

Then you can easily construct a computation object within your test with custom parameters.